src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlArrayMethodCallTranslator.cs

```diff
@@ -0,0 +1,401 @@
+        {
+            if (!(expression.Object is Expression instance))
+                return null;
+
+            if (!instance.Type.IsArray)
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case nameof(Array.GetLength) when VersionAtLeast(8, 4):
+                return
+                    new SqlFunctionExpression(
+                        "array_length",
+                        typeof(int),
+                        new[] { instance, GenerateOneBasedIndexExpression(expression.Arguments[0]) });
+
+            case nameof(Array.GetLowerBound) when VersionAtLeast(7, 4):
+                return
+                    Coalesce(
+                        new SqlFunctionExpression(
+                            "array_lower",
+                            typeof(int),
+                            new[] { instance, GenerateOneBasedIndexExpression(expression.Arguments[0]) }));
+
+            case nameof(Array.GetUpperBound) when VersionAtLeast(7, 4):
+                return
+                    Coalesce(
+                        new SqlFunctionExpression(
+                            "array_upper",
+                            typeof(int),
+                            new[] { instance, GenerateOneBasedIndexExpression(expression.Arguments[0]) }));
+
+            default:
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Translates instance methods defined on <see cref="List{T}"/>.
+        /// </summary>
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        Expression ListInstanceHandler([NotNull] MethodCallExpression expression)
+        {
+            if (!(expression.Object is Expression instance))
+                return null;
+
+            if (!instance.Type.IsGenericType || instance.Type.GetGenericTypeDefinition() != typeof(List<>))
+                return null;
+
+            switch (expression.Method.Name)
+            {
+            case "get_Item":
+                return MakeIndex(instance, expression.Arguments[0]);
+
+            case nameof(IList.IndexOf) when VersionAtLeast(9, 5):
+                return Coalesce(new SqlFunctionExpression("array_position", typeof(int), new[] { instance, expression.Arguments[0] }));
+
+            default:
+                return null;
+            }
+        }
+
+        /// <summary>
+        /// Translates instance methods defined on <see cref="string"/>.
+        /// </summary>
+        /// <param name="expression">The expression to translate.</param>
+        /// <returns>
+        /// A translated expression or null if no translation is supported.
+        /// </returns>
+        [CanBeNull]
+        static Expression StringInstanceHandler([NotNull] MethodCallExpression expression)
```

Comment:
As written above, it may be better to move string translations to their own translator.

Reply:
I'm open to moving this... but it would require some duplication in the `Enumerable` and index handling (see the next comment).

I'm going to hold off moving anything in the next commit, but still open to it if you think the duplication trade-off is acceptable.

Reply:
Yeah, looking at this again after a while I'd move it to a string translator. It makes more sense to me to have a translator per CLR type (roughly), much easier to discover who does which translations etc.

Reply:
Okay, sounds good. It fits nicely into the refactored `NpgsqlStringTranslator` (which I don't think we had back in August).